File: src/main/java/com/example/trainingsystem/models/Account.java
package com.example.trainingsystem.models;

import com.example.trainingsystem.enums.ERole;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import javax.validation.constraints.NotBlank;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Document(collection = "Accounts")
public class Account {
    @Id
    @JsonProperty("_id")
    private String _id;

    @NotBlank
    @Indexed(unique = true)
    @JsonProperty("Username")
    private String Username;

    @NotBlank
    @JsonProperty("Password")
    private String Password;

    @Field
    @JsonProperty("Role")
    private ERole Role;
}

File: TrainingSystem.NetAPI/MongoDB_CRUD/Models/Account.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Collections.Generic;
using System;


namespace MongoDB_CRUD.Models
{
    public class Account
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        [BsonElement("Username")]
        public string Username { get; set; }
        [BsonElement("Password")]
        public string Password { get; set; }
        [BsonElement("Role")]
        public string Role { get; set; }
        [BsonElement("_class")]
        public string _class { get; set; }




    }
}
